root/static/scripts/common/utility/deepEqual.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * @flow
+ * Copyright (C) 2020 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+const objectPrototype = Object.prototype;
+const hasOwnProperty = objectPrototype.hasOwnProperty;
+
+export default function deepEqual(
+  a: any,
+  b: any,
+  cmp?: (any, any) => boolean,
```

Comment:
Any chance we can make them not `any` and this flow strict?

Reply:
Not really, for the same reason implementing `clone` is hard without `any`. Flow refinement isn't good enough yet to support the kind of magic this does.

Reply:
But luckily we shouldn't be needing to use this function much (outside of tests) in Reactified code, since we won't need to deeply-compare state objects there.